tests/framework/extensions/clusters/clusters.go

```diff
@@ -162,6 +155,49 @@ func NewRKE2ClusterConfig(clusterName, namespace, cni, cloudCredentialSecretName
 	return v1Cluster
 }
 
+// CreateRKE1Cluster is a "helper" functions that takes a rancher client, and the rke1 cluster config as parameters. This function
+// registers a delete cluster fuction with a wait.WatchWait to ensure the cluster is removed cleanly.
+func CreateRKE1Cluster(client *rancher.Client, rke1Cluster *management.Cluster) (*management.Cluster, error) {
+	opts := metav1.ListOptions{
+		FieldSelector:  "metadata.name=",
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	}
+
+	cluster, err := client.Management.Cluster.Create(rke1Cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	client.Session.RegisterCleanupFunc(func() error {
+		err := client.Management.Cluster.Delete(rke1Cluster)
+		if err != nil {
+			return err
+		}
+
+		adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
+		if err != nil {
+			return err
+		}
+
+		watchInterface, err := adminClient.GetManagementWatchInterface(management.ClusterType, opts)
+
+		if err != nil {
+			return err
+		}
+
+		return wait.WatchWait(watchInterface, func(event watch.Event) (ready bool, err error) {
+			if event.Type == watch.Error {
+				return false, fmt.Errorf("there was an error deleting cluster")
```

Comment:
is the error returned from `watch` something that could help with debugging why the cluster isn't deleted properly?

Reply:
What do you mean Caleb? WatchWait does return an error if there is indeed one, so 
`return wait.WatchWait()` is the same as 


`err = watit.WatchWait()`
`return err`

Reply:
oh ok got it, thanks!